### DISCUSSION TITLE
rdio-scannner Munin Stats Plugin

### DISCUSSION THREAD

**Discussion by hayden-t** [chuot/rdio-scanner#139] [Show and tell]
Here is a python3 based script to graph trunk player stats:

A guide to install the plugin/munin can be inferred from my Trunk Recorder stats plugin page [link](https://github.com/robotastic/trunk-recorder/wiki/Monitoring-system-performance,-activity-&-email-alerts)
You will also find details there on setting up optional email alerts.

![rdioscanner_munin_py-pinpoint16465743261646682326](https://user-images.githubusercontent.com/1000743/157107968-614ec6aa-c6f1-4c39-859d-534f32591616.png)

```
#!/usr/bin/python3
import subprocess
import sys

title = "Rdio-Scanner Stats per 5min"
timeSpan = '5 minutes ago'

if('config' in sys.argv):
	print('graph_title '+title)
	print('graph_vlabel Count')
	print('graph_printf %.0lf' )	
	print('graph_category rdioscanner')
	print('value1.label Listeners')
	print('value2.label Valid')
	print('value3.label Failed Login')
	print('value4.label Duplicate Calls')
	print('value5.label Missing TG')
	sys.exit()

proc = subprocess.run(['journalctl', '--no-pager', '-u', 'rdio-scanner.service', '--since', timeSpan], capture_output=True, text=True)

invalidCode = duplicateCalls = missingTG = validTG = listeners = 0

### temp storage between runs, incase next run doesnt have listener count
f = open("/tmp/muninRdioScanner.txt", "a+")
f.seek(0)
listeners = f.read()
if not listeners:
	listeners = 0
###

for line in proc.stdout.splitlines():
	if 'listeners count is' in line:#how often listed ?
		listeners = line.split()[-1]
	if 'invalid access code' in line:
		invalidCode+=1
	if 'duplicate call rejected' in line:
		duplicateCalls+=1
	if 'no matching system' in line:
		missingTG+=1
	if 'success' in line:
		validTG+=1
		
		
print("value1.value "+str(listeners))
print("value2.value "+str(validTG))
print("value3.value "+str(invalidCode))
print("value4.value "+str(duplicateCalls))
print("value5.value "+str(missingTG))


### update temp storage
f.seek(0)
f.truncate()
f.write(str(listeners))
f.close()
###
```

**Comment by chuot** [comment]
That is great, thanks for sharing this.

**Reply by hayden-t** [reply]
Thanks, Ive just added a screenshot of graph, and corrected the title

**Reply by jslitzkerttcu** [reply]
@hayden-t  - this is great!  Got it installed.  Do you have code you would be willing to share for the template page to display the graph?  I can see the stats in the CLI just not in the apache UI.  I'm assuming I need to generate a template page?

**Reply by hayden-t** [reply]
It just appears in Apache graphs automatically,  do you have other graphs generating OK?

**Reply by hayden-t** [reply]
The python script above supplies everything in the right format for munin to generate graphs

**Reply by jslitzkerttcu** [reply]
My apologies, I think I just didn't wait long enough.  You are correct, graphs are generating correctly now.  Thank you for sharing the script and detailed information on how to configure it!

**Reply by hayden-t** [reply]
Yeah that has got me a few times too with munin, you dont realise its just a matter of waiting for the graph to generate. There are commands you can use to force an update that ive used when developing but that is another topic. Glad it works for you. Munin is a very easy system to graph for, so you should be able to easily modify my script if needed.

**Comment by LoveScanner** [comment]
It will only work in Linux right?

**Reply by hayden-t** [reply]
yeah looks like munin is linux only, but i cant more highly recommend learning linux as a skill

**Reply by LoveScanner** [reply]
I know Linux.

The question is what will this give me?

I was happy to switch from sdrtrunk to trunk-recorder linux (I also tried)
The problem is I felt it worked slower and less well.

And for Sdrtrunk it will probably work the same Win & linux

**Reply by hayden-t** [reply]
sdrtrunk is very pretty and easy to use, one reason people choose trunk recorder because it can run on small embedded linux systems such as rpi or odroid n2 and not take up much space, power, or noise for 24/7 operation and streaming

**Reply by LoveScanner** [reply]
From my experience sdrtrunk records faster once you need recording from ten sites in parallel
If you think otherwise I am ready to try again :)

**Comment by hayden-t** [comment]
It just appears in Apache graphs automatically,  do you have other graphs generating OK?
-------- Original message --------From: jslitzkerttcu ***@***.***> Date: 9/6/22  9:52 am  (GMT+10:00) To: chuot/rdio-scanner ***@***.***> Cc: hayden-t ***@***.***>, Mention ***@***.***> Subject: Re: [chuot/rdio-scanner] rdio-scannner Munin Stats Plugin (Discussion #139) 
@hayden-t  - this is great!  Got it installed.  Do you have code you would be willing to share for the template page to display the graph?  I can see the stats in the CLI just not in the apache UI.  I'm assuming I need to generate a template page?

**Reply by jslitzkerttcu** [reply]
Did you mean to post this?  It looks like the same reply you sent to me back in June where we resolved it?

**Comment by theaustinayers** [comment]
Any way to get this to integrate into Observium, maybe SNMP?

**Reply by hayden-t** [reply]
sorry, maybe you can :)
